sandbox-tools/merge-repos/src/syncReposToStaging.ts

```diff
@@ -507,100 +509,222 @@ async function resolveConflictsToTheirs(git: SimpleGit, commitDetails: ICommitDe
+                        await git.rm(destFolder + "/" + theFile)
+    
+                        fs.mkdirSync(destFolder + "/" + theFile);
+                        await checkFixBadMerges(git, repoName, baseFolder + "/" + theFile, destFolder + "/" + theFile, commitDetails, level + 1);
+                    }
                 } else {
-                    log(` - ${fullInputPath}`);
+                    log(` - (Missing) ${baseFolder} <=> ${destFolder}`);
+                    fs.mkdirSync(destFolder + "/" + theFile);
+                    await checkFixBadMerges(git, repoName, baseFolder + "/" + theFile, destFolder + "/" + theFile, commitDetails, level + 1);
                 }
+            } else {
+                // Assume file
+                if (validateFile(baseFolder + "/" +  theFile,  destFolder + "/" +  theFile)) {
 
-                if (!moved) {
                     await git.raw([
-                        "mv",
-                        "--force",
-                        "--verbose",
-                        fullInputPath + (isSrcDir ? "/" : ""),
-                        theGitDestFolder + (isSrcDir ? "/" + inputFile + "/" : "")
-                    ]);
-
-                    commitMessage = appendCommitMessage(commitMessage, ` - ${fullInputPath}${isSrcDir ? "/" : ""}`)
+                        "add",
+                        "-f",
+                        destFolder + "/" +  theFile]);
                 }
+            }
+        } else {
+            log(` - (xI) ${baseFolder + "/" + masterFiles[mLp]} ==> ${theFile}`);
+        }
+    }
+
+    // Remove any files / folders that do not exist in the master repo
+    for (let dLp = 0; dLp < destFiles.length; dLp++) {
+        let destFile = destFiles[dLp];
+        if (masterFiles.indexOf(destFile) === -1 && !_isVerifyIgnore(destFile, true)) {
+            let destStats = fs.statSync(destFolder + "/" + destFile);
+            if (destStats.isDirectory()) {
+                commitDetails.message = logAppendMessage(commitDetails.message, { index: "*", working_dir: "F", path: destFolder + "/" + destFile } , `Removing extra folder ${destFile}`);
+                await git.raw([
+                    "rm",
+                    "-f",
+                    "-r",
+                    destFolder + "/" + destFile]);
+                // fs.rmdirSync(destFolder + "/" + destFile, {
+                //     recursive: true
+                // });
             } else {
-                log(` - Ignoring ${inputFile}  (${destFolder})`);
+                commitDetails.message = logAppendMessage(commitDetails.message, { index: "*", working_dir: "E", path: destFolder + "/" + destFile } , "Removing extra file");
+                await git.rm(destFolder + "/" + destFile);
             }
-        };
+        }
+    }
+}
+
+async function getAndSyncSrcRepo(git: SimpleGit, repoName: string, details: IRepoDetails): Promise<ISourceRepoDetails> {
```

Comment:
New sequence that `clones` the original `source` repo locally instead of just adding as a `remote`.

The previous sequence used the `remote` to merge directly into the `staged` (this caused a lot of the incorrect matching).

The new sequence create a `clone` and then moves it around locally, before using this local clone as the `remote` for the `git merge`, this reduces (but does not completly avoid) git incorrectly matching files / folders between the JS and CONTRIB repos.